File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.cpp
class Solution {
   public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n, 1);
        int prod = 1;
        for (int i = 1; i < n; i++) {
            res[i] = prod * nums[i - 1];
            prod = res[i];
        }
        prod = 1;
        for (int i = n -1; i >= 0; i--) {
            res[i] *= prod;
            prod*=nums[i];
        }
        return res;
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None:
            return []
        res = []
        q = collections.deque()
        q.append(root)

        while q:
            nq=[]
            level=[]
            
            for node in q:
                level.append(node.val)
                if node.left:
                    nq.append(node.left)
                if node.right:
                    nq.append(node.right)
            res.append(level)
            q=nq 
            
        return res
        